lib/manager/components/transform/SubstitutionRow.js

```diff
@@ -29,22 +29,30 @@ const buttonGroupStyle = { width: '60px' }
  * Extract just the subset of fields to build the state variable.
  */
 function extractStateFields (props: Props): Substitution {
-  const { description, normalizeSpace, pattern, replacement } = props.substitution
+  const { description, normalizeSpace, pattern, replacement, valid } = props.substitution
   return {
     description: description || '',
-    normalizeSpace,
+    normalizeSpace: normalizeSpace || false,
     pattern,
-    replacement
+    replacement,
+    valid
   }
 }
 
 /**
  * @returns true if the given substitution pattern is null or empty, false otherwise.
  */
-export function isSubstitutionInvalid (substitution: Substitution) {
+export function isSubstitutionBlank (substitution: Substitution) {
   return !substitution.pattern || substitution.pattern === ''
 }
 
+/**
+ * @returns true if the given substitution is marked invalid by the backend, false otherwise.
+ */
+export function isSubstitutionInvalid (substitution: Substitution) {
+  return !substitution.valid
+}
```

Comment:
To better organize things, can these util functions be moved to a util file?

Reply:
Actually, maybe just the `isSubstitutionBlank` and `isSubstitutionInvalid` functions.

Reply:
Updated in https://github.com/ibi-group/datatools-ui/pull/687/commits/cf0d3a024691f9841665e88aba94cd027048044b.